src/main/java/it/pagopa/pagopa/apiconfig/service/IcaService.java

```diff
@@ -58,4 +77,53 @@ public Icas getIcas(@NotNull Integer limit, @NotNull Integer pageNumber) {
                 .map(elem -> modelMapper.map(elem, Ica.class))
                 .collect(Collectors.toList());
     }
+
+    public XSDValidation verifyXSD(File xml) {
+        boolean xsdEvaluated = false;
+        String lineNumber = "";
+        String detail;
+        String xsdSchema = xsdProperties.getIca();
+
+        try {
+            SchemaFactory factory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI);
+            // to be compliant, prohibit the use of all protocols by external entities:
+            factory.setProperty(XMLConstants.ACCESS_EXTERNAL_DTD, "");
+            factory.setProperty(XMLConstants.ACCESS_EXTERNAL_SCHEMA, "");
+
+            javax.xml.validation.Schema schema = factory.newSchema(new URL(xsdSchema));
+            Validator validator = schema.newValidator();
+
+            XMLInputFactory xmlInputFactory = XMLInputFactory.newInstance();
+            // to be compliant, completely disable DOCTYPE declaration:
+            xmlInputFactory.setProperty(XMLInputFactory.SUPPORT_DTD, false);
+
+            FileInputStream inputStream = new FileInputStream(xml);
+            XMLStreamReader xmlStreamReader = xmlInputFactory.createXMLStreamReader(inputStream);
```

Comment:
Why use storageService? I think it introduces a lot of complexity. If you need to convert MultipartFile into inputStream see [here](https://www.baeldung.com/spring-multipartfile-to-file).